src/ui/public/kuery/lib/functions/not.js

```diff
@@ -0,0 +1,38 @@
+import * as ast from '../ast';
+import { nodeTypes } from '../node_types';
+
+export function buildNodeParams(child, serializeStyle = 'operator') {
+  return {
+    arguments: [child],
+    serializeStyle
+  };
+}
+
+export function toElasticsearchQuery(node, indexPattern) {
+  let [ argument ] = node.arguments;
+  if (argument.type === 'literal') {
+    argument = nodeTypes.function.buildNode('is', '*', argument.value);
+  }
+
+  return {
+    bool: {
+      must_not: ast.toElasticsearchQuery(argument, indexPattern)
+    }
+  };
+}
+
+export function toKueryExpression(node) {
+  if (node.serializeStyle !== 'operator') {
+    throw new Error(`Cannot serialize "not" function as "${node.serializeStyle}"`);
+  }
+
+  const [ argument ] = node.arguments;
+  const queryString = ast.toKueryExpression(argument);
+
+  if (argument.function && (argument.function === 'and' || argument.function === 'or')) {
```

Comment:
It might be less coupled to `and` and `or` specifically if this looked at the serialization style to decide whether it is an infix that needs grouping or whether it is a "self-grouping" prefix style expression. That way it would also leave out the parentheses with prefix-style `and`s and `or`s.